tests/unit/cache.test.ts

```diff
@@ -0,0 +1,24 @@
+import { assert } from "deno/testing/asserts.ts"
+import { useDownload } from "hooks"
+import { sandbox } from "../utils.ts"
+
+Deno.test("etag-mtime-check",async () => {
+  await sandbox(async tmpdir => {
+    const src = new URL("https://dist.tea.xyz/deno.land/darwin/aarch64/v1.27.1.tar.xz")
```

Comment:
This is a 30MB file, pick something less than 100B pls.

Also, this is fragile. We should not rely on the Internet for testing. Really you should spin up a local web server using a tea-script and test against that.

Reply:
the work required for that test infrastructure is pretty high tho. However tests failing due to flakey network conditions is the worst.

However, it will pay off. We can use it to test just about everything.

Reply:
It might be as simple as `tea -X npx http-server test.fixtures/ -p 2051`

Caveats:
- 2051 = the ordinals for T(20) E(5) A(1)
- 2051 is `epnsdp`, which a quick google search might be an archaic dns thing, and used by some commercial backup solutions in their products (the port, not the service)
- the `tea -X npx` command will *also* flake if there's internet issues, but it allows a single install call before any number of http tests